src/uiSelectController.js

```diff
@@ -198,12 +200,29 @@ uis.controller('uiSelectCtrl',
       ctrl.groups.forEach(function(group) {
         ctrl.items = ctrl.items.concat(group.items);
       });
+      addNullItem();
```

Comment:
We should only add the null item if `allow-clear=true` or `ng-required=false` depending on the implementation.